Makefile

```diff
@@ -48,27 +56,25 @@ clean-dockerdemo:
 ssh:
 	@vagrant ssh netplugin-node1 || echo 'Please run "make demo"'
 
-unit-test: build
-	CONTIV_HOST_GOPATH=$(GOPATH) CONTIV_HOST_GOBIN=$(HOST_GOBIN) \
-					   CONTIV_HOST_GOROOT=$(HOST_GOROOT) ./scripts/unittests -vagrant
+unit-test: build clean
```

Comment:
Let me pour over this today and see if I can rectify things.

-Erik



On August 10, 2015 at 3:06:12 PM, Madhav Puri (notifications@github.com) wrote:

In Makefile:

> @@ -48,27 +56,25 @@ clean-dockerdemo:
>  ssh:
>    @vagrant ssh netplugin-node1 || echo 'Please run "make demo"'
>   
> -unit-test: build
> -  CONTIV_HOST_GOPATH=$(GOPATH) CONTIV_HOST_GOBIN=$(HOST_GOBIN) \
> -                                     CONTIV_HOST_GOROOT=$(HOST_GOROOT) ./scripts/unittests -vagrant
> +unit-test: build clean
I see, so essentially the clean forces a rebuild on host? It might then fail for MAC, right?

Previously this was solved by having CONTIV_HOST_GOBIN and CONTIV_HOST_GOROOT vars, which would mount host's go environment inside the vm and avoid the mismatch of the obj files no matter where the build is triggered.

I didn't realize we still had this issue since system tests are being built natively on the host, may be we can keep the instrumentation of CONTIV_HOST_GOBIN and CONTIV_HOST_GOROOT intact.

—
Reply to this email directly or view it on GitHub.